lib/install/management/appliance.go

```diff
@@ -67,14 +68,42 @@ func (d *Dispatcher) isVCH(vm *vm.VirtualMachine) (bool, error) {
 	return false, nil
 }
 
-func (d *Dispatcher) checkExistence(conf *metadata.VirtualContainerHostConfigSpec) error {
+func (d *Dispatcher) checkExistence(conf *metadata.VirtualContainerHostConfigSpec, settings *data.InstallerData) error {
 	defer trace.End(trace.Begin(""))
 
-	vm, err := d.findAppliance(conf)
+	var err error
+	d.vchPoolPath = fmt.Sprintf("%s/%s", settings.ResourcePoolPath, conf.Name)
+	var orp *object.ResourcePool
+	var vapp *object.VirtualApp
+	if d.isVC {
+		vapp, err = d.findVirtualApp(d.vchPoolPath)
+		if err != nil {
+			return err
+		}
+		if vapp != nil {
+			orp = vapp.ResourcePool
+		}
+	}
+	if orp == nil {
+		if orp, err = d.findResourcePool(d.vchPoolPath); err != nil {
+			return err
+		}
+	}
+	if orp == nil {
+		return nil
+	}
+
+	rp := compute.NewResourcePool(d.ctx, d.session, orp.Reference())
+	vm, err := rp.GetChildVM(d.ctx, d.session, conf.Name)
 	if err != nil {
 		return err
 	}
 	if vm == nil {
+		if vapp != nil {
+			err = errors.Errorf("virtual app %s is found, but is not VCH, please choose different name", d.vchPoolPath)
```

Comment:
Would be good in general if we use `%q` rather than `%s`, the quotes help to see if user input happens to include whitespace.